packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -57,6 +90,115 @@ export class SnapshotReporter {
     );
   }
 
+  private reportHandlers: SnapshotReporterHandlers = {
+    [SnapshotReportStatus.SUCCESS]: () => {},
+    [SnapshotReportStatus.NO_CHANGES]: () => {},
+    [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: () => {},
+    [SnapshotReportStatus.ERROR]: () => {},
+  };
+
+  public setReportHandler<T extends SnapshotReportStatus>(
+    status: T,
+    handler: SnapshotReporterHandlers[T]
+  ): SnapshotReporter {
+    this.reportHandlers[status] = handler;
+    return this;
+  }
+
+  public async handleReport(): Promise<void> {
+    const reportStatuses = this.getReportStatuses();
+    for (const handler of reportStatuses.fixables) {
```

Comment:
nit: Could remove some repetition and improve readability

```typescript
public async handleReport(): Promise<void> {
  const reportStatuses = this.getReportStatuses();
   // or extract a type for what's returned by `getReportStatuses` top level in this file
   type reportStatusProp = keyof typeof reportStatuses;
  
  const executeHandlers = async (k: reportStatusProp) => {
     for (const handler of reportStatuses[k]) {
          await this.callAndReset(handler);
        }
  }

  const has = (k: reportStatusProp) => {
       return reportStatuses[k].length > 0;
  }
  
  await executeHandlers('fixables');
  if (has('errors')) {
      await executeHandlers('errors');      
      return;
  }

  if (has('successes')) {
      await executeHandlers('successes');      
  }
}
```

Reply:
I took that and tweaked it a bit.
I'm not found of using string as keys, it's a pain to refactor, likewise, I think has is a bit overkill. but the 'spirit' of the comment has been followed.